src/test/java/org/commcare/formplayer/services/MediaMetaDataServiceTest.java

```diff
@@ -102,39 +113,68 @@ public List<MediaMetadataRecord> answer(InvocationOnMock invocation) throws Thro
 
     @AfterEach
     public void tearDown() {
+        cacheManager.getCache(MEDIA_METADATA_CACHE).clear();
         testFile.delete();
     }
 
     @Test
-    public void testSaveMetaData() {
+    public void testMediaMetadataCacheExists() {
+        Cache cache = cacheManager.getCache(MEDIA_METADATA_CACHE);
+        assertNotNull(cache);
+        assertTrue(cache instanceof CaffeineCache);
+    }
+
+    @Test
+    public void testSaveMetaData_cached() {
+        cacheManager.getCache(MEDIA_METADATA_CACHE).clear();
+
         mediaMetaDataService.saveMediaMetaData(mediaMetaData);
-        Optional<MediaMetadataRecord> fetchedMediaMetaData = mediaMetaDataRepo.findById(mediaMetaData.getId());
-        Assertions.assertTrue(fetchedMediaMetaData.isPresent());
+
+        assertEquals(mediaMetaData, getCachedMetadata(mediaMetaData.getId()).get());
+
+        MediaMetadataRecord fetchedMediaMetaData = mediaMetaDataService.getMetadataById(mediaMetaData.getId());
```

Comment:
Ahh sorry I didn't mean for us to introduce a new function in service layer that we are only using in test. Since we are querying the service in live code by using `findByFileId` we should be using that method to get the record here. And also while writing the record, instead of caching by `#mediaMetadataRecord.id` we should cache by `fileId` since that's what we are using to query the DB later on.

Reply:
That makes total sense. I wasn't thinking about the broader implications. I switched to using `findByFileId` and caching with `fileId`.